### PR TITLE
[Automated Guidelines] Add index, template, links

### PR DESCRIPTION
**Pull request by RobDolinMS**
Add:
* Index section
* Template for what an example issue details section could look like 
* Links at bottom of template section back to top sections in rest of document

Also fix typo: "warningss" --> "warnings"

Tagging @kirthik to take a look.  Proposed edits are nicely visible here: https://github.com/RobDolinMS/azure-rest-api-specs/blob/patch-3/documentation/openapi-authoring-automated-guidelines.md

Signed-off-by: Rob Dolin <robdolin@microsoft.com>

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes. - n/a
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec). - n/a

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- Other topics in this section were not applicable since no OpenAPI (Swagger) files are being edited by this PR.

### CODE REVIEW

File: documentation/openapi-authoring-automated-guidelines.md [hunk 1]
```diff
@@ -94,3 +104,30 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
 | S2004	| NonAppJsonTypeWarning | Please make sure that media types other than 'application/json' are supported by your service. | Warning |
 | M2063	| BodyParametersValidation | A body parameter must be named 'parameters'. | Warning |
 | M2017	| PutRequestResponseValidation | A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. Operation: '{0}' Request Model: '{1}' Response Model: '{2}' | Warning |
+
+
```
Comment (kirthik) [inline documentation/openapi-authoring-automated-guidelines.md]: Can you please add a heading here - Rule descriptions or something like that and index it at the top?

File: documentation/openapi-authoring-automated-guidelines.md [hunk 2]
```diff
@@ -94,3 +104,30 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
...
+## <a name="M3012" />M3012 APIVersionPattern
+**Output Message**: API Version must be in the format: yyyy-MM-dd, optionally followed by -preview, -alpha, -beta, -rc, -privatepreview.
+
+**Description**: The API Version paramemter MUST be in the Year-Month-Date format (i.e. 2016-07-04.)  NOTE that this is the en-US ordering of month and date.  
+
+They date MAY optionally be followed by one of:
```
Comment (kirthik) [inline documentation/openapi-authoring-automated-guidelines.md]: The data MAY...

File: documentation/openapi-authoring-automated-guidelines.md [hunk 3]
```diff
@@ -94,3 +104,30 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
...
+* -preview - Indicates the API version is in (public) preview
+* -alpha
+* -beta
+* -rc (release candidate)
+* -privatepreview
+
+**Good Examples**: Examples of valid version patterns include:
+* 2016-07-04
+* 2016-07-04-beta
```
Comment (kirthik) [inline documentation/openapi-authoring-automated-guidelines.md]: 2016-07-04-preview. I did not see any beta tags until now. -preview is very common

**Comment by RobDolinMS** [comment]
Thanks @kirthik I believe I've fixed the items you commented on.  If so, please give this a thumbs-up.  Thanks.

**Review by kirthik — APPROVED** [review]

**Comment by kirthik** [comment]
:shipit: from me. @sergey-shandar - Changes look good to me.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**